src/Language/ASKEE/DataSet.hs

```diff
@@ -0,0 +1,97 @@
+{-# LANGUAGE OverloadedStrings #-}
+module Language.ASKEE.DataSet
+( DataSet(..)
+, DataSetColumn(..)
+, dataSetFromDataSeries
+) where
+
+import qualified Data.Aeson as JSON
+import Data.Aeson((.=))
+import qualified Data.Aeson.Types as JSONT
+import qualified Data.HashMap.Lazy as HM
+import Data.Text(Text)
+import qualified Data.Text as Text
+import qualified Data.Map as Map
+
+import qualified Language.ASKEE.DataSeries as Series
+import Language.ASKEE.DataSeries(DataSeries)
+
+-- DataSet is a data format that includes some metadata
+data DataSet = DataSet
+  { dataSetName :: Text
+  , dataSetDescription :: Maybe Text
+  , dataSetColumns :: [DataSetColumn]
+  }
+  deriving(Show, Eq, Ord)
+
+data DataSetColumn = DataSetColumn
+  { dataSetColumnName :: Text
+  , dataSetColumnDescription :: Maybe Text
+  , dataSetColumnValues :: [Double]
+  }
+  deriving(Show, Eq, Ord)
```

Comment:
Do you think it's worth attempting to unify this with `DataSeries`? As a structure it feels a bit duplicative, but `DataSeries` also feels a bit ingrained.

Reply:
I considered it, we'd have to do a lot to `DataSeries` to make it handle metadata correctly and it would involve touching a bunch of stuff in the process which I would prefer not to do.  I think this is fine as a "DataSeries with metadata that doesn't eseteem `time` specially" although I wonder if the columns should be a map

Reply:
In Haskell, or in JSON, or both? I have no qualms with changing the Haskell but I slightly prefer the current JSON schema to, say: 
```json
{
  "columns": [
    {
      "S": { ... },
      "I": { ... },
      ...
    }
  ]
}
```

Reply:
just the Haskell representation, i think i like the schema as it is also - wrangling keys in JSON is kind of a pain

Reply:
Yeah, no real opposition, up to you. I suppose it might be nice to have the JSON structure mirror the Haskell, but I do love a good `Map`.